removed other layouts tests for win32

gulnazsayed opened the pull request:
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Removed "other" layouts (horizontal and horizontal-stacked) from win32 tester app. Added a comment in the spec and to the layout and labelPosition props that these layouts are not supported in win32. We decided that these "other" layouts are not really seen in win32 so it is better not to expose these layouts to partners until a use case is identified specifically for the win32 platform. The logic and props are left as is for other platforms to use and in the case we want to make it available for win32 in the future.

### Verification

Manually tested that the horizontal and horizontal-stacked layout tests are removed from the win32 tester app.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

rurikoaraki commented:
Are you also removing the horizontal layout capability from the radio group? This change doesn't seem to restrict anyone from using the props.

gulnazsayed commented:
> Are you also removing the horizontal layout capability from the radio group? This change doesn't seem to restrict anyone from using the props.

We want to keep the props so that the other platforms can use them and/or we can add the functionality back easily if we identify a use case later down the line. Android doesn't support these layouts either, which seems to only be enforced by the props' comments and not exposing it in the platform's tester app. 

In the latest commit, I added a check for win32 to enforce that the layout and labelPosition props are always the default. I added one for android in useRadioGroup as well.

rurikoaraki commented:
> 

We're only shipping this so far for Android and win32 right? My opinion would be that it's better to take them out and let someone add them back in if they need it for another platform. It would still be an extra code to have to upkeep if we leave it in even if nobody uses it right now.

But if you want to leave it in, the way to enforce it in win32/android would be to probably force the layer to always resolve to a different option if one of the horizontal layouts is picked. If it's not supported on the win32 or android platforms we shouldn't be letting the control resolve to unsupported layouts.

gulnazsayed commented on packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:
```diff
@@ -100,7 +102,7 @@ export const useRadioGroup = (props: RadioGroupProps): RadioGroupInfo => {
       ...props,
       required,
-      layout,
+      layout: layout,
```
Fixed and confirmed it works as expected

rurikoaraki commented on packages/experimental/RadioGroup/src/Radio/useRadio.win32.ts:
```diff
@@ -29,7 +29,7 @@ export const useRadio = (props: RadioProps): RadioInfo => {
     value,
     disabled,
-    labelPosition = radioGroupContext.layout === 'horizontal-stacked' ? 'below' : 'after',
+    labelPosition = 'after', // disables labelPosition in win32
```
I think this should be a hardcoded const. If we take it from props someone can override it by passing a different value

gulnazsayed replied:
Fixed, thanks for the comment. We are shipping it only on win32 and android for now, but I decided to keep the horizontal layout code so we can easily re-enable it later if needed to align with Fluent. I changed the fix to force the layers to resolve the horizontal layouts to the default vertical, like you suggested.

rurikoaraki commented on packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { Platform } from 'react-native';
+// import { Platform } from 'react-native';
```
nit: remove?

gulnazsayed replied:
Removed!
